jenkins/docker/docker-build.jenkinsfile

```diff
@@ -40,18 +35,15 @@ pipeline {
             }
             steps {
                 script {
+                    echo 'The docker-build workflow will only push docker images to staging, please use docker-copy to move the image to other repositories'
                     git url: "$DOCKER_BUILD_GIT_REPOSITORY", branch: "$DOCKER_BUILD_GIT_REPOSITORY_REFERENCE"
-                    def CREDENTIAL_ID = "jenkins-staging-docker-staging-credential"
-                    if (env.IS_STAGING == "false") {
-                        CREDENTIAL_ID = "jenkins-staging-docker-prod-token"
-                        sh "echo Switch to Production"
-                    }
+                    def CREDENTIAL_ID = "jenkins-staging-dockerhub-credential"
                     sh "echo Account: $CREDENTIAL_ID"
                     withCredentials([usernamePassword(credentialsId: CREDENTIAL_ID, usernameVariable: 'DOCKER_USERNAME', passwordVariable: 'DOCKER_PASSWORD')]) {
                         sh '''
-                            set -ex
+                            set -e
                             echo Login to $CREDENTIAL_ID
-                            docker logout && docker login -u $DOCKER_USERNAME -p $DOCKER_PASSWORD && eval $DOCKER_BUILD_SCRIPT_WITH_COMMANDS
+                            docker logout && echo $DOCKER_PASSWORD | docker login -u $DOCKER_USERNAME --password-stdin && eval $DOCKER_BUILD_SCRIPT_WITH_COMMANDS
```

Comment:
Echoing password is not a good practice. Is this change intended?

Reply:
@prudhvigodithi 
This is intended and it is actually much safer than the original setup.
Note that the password is pipe directly into --password-stdin instead of stdout.
In that case `docker login` command will not print `login with -u **** -p ****` which is the best practice from docker.

Reply:
But when you do echo $DOCKER_PASSWORD it will emit an password right?

Reply:
It send the password to the pipe it did not emit to anywhere else.

Reply:
Okay still seems risky. Better to monitor and maybe enhance later.

Reply:
Why can't we keep what was before? I am not comfortable with echoing password.

Reply:
[Offline discussion] Looks like previously docker did emit the password. This way it will not. Hence, this is better. Thanks!

Reply:
I see we also use it here. https://github.com/opensearch-project/opensearch-build/blob/92efa806f8bb3321aa0af40711ba6de3b879b93f/vars/copyContainer.groovy#L16

Reply:
Thank you!